rfcs/2022-11-04-automatic-component-validation.md

```diff
@@ -410,17 +379,9 @@ leading to a permanent trail of subtly-wrong internal telemetry.
 
 ## Drawbacks
 
-The main drawback of this work is the additional boilerplate required for each component. While not
-particularly onerous, it does represent another bit of code that looks and feels a lot like
-boilerplate that already required, such as `SourceConfig`, and so on.
-
-Additionally, this proposal does not (at least not yet) provide any consideration to
-testing/validating components under a series of varying inputs -- i.e. sending all metrics vs
-sending all logs, changing an aspect of the component's configuration itself, etc -- which means
-that updating it to do so in the future could potentially be hard work, as all components may need
-to be slightly refactored to deal with changes to the core traits, and so on. This is relevant to
-the design, as it stands, having no clear way to automatically derive the implementation of the
-proposed trait from existing data, forcing all implementations to be written manually.
+The main drawback of this work is the additional data required per component that is not coupled to
+source code: this represents another potential avenue of drift between the component as it is
+implemented/described in Rust source vs how it is described in the test configuration data.
```

Comment:
:thought_balloon:
I've re-read the discussion between you and @lukesteensen regarding the more external approach and the more code/compile time driven approach a couple times now.
Despite reading that (and perhaps I'm not fully comprehending something about either approach but), and seeing the value of the external approach providing more simplicity, I'm concerned the external approach outlined here leaves too much room for the "drift" you mention here, to turn into validation gaps and consequently incorrectness going undetected.
It _feels_ like the more embedded approach, having the components implement traits, puts the importance of validation more "in your face" so to speak, and get's more to the heart at facing the issue's we've had historically in this area. Spencer mentioned the external approach is less invasive which is true, but maybe it needs to be more invasive to accomplish the goal?

If it's an external process with those data files needing to be kept up, how will we go about keeping it accurate?